habitat-lab/habitat/tasks/rearrange/utils.py

```diff
@@ -590,21 +605,22 @@ def _get_robot_spawns(
         if np.isnan(propose_position).any():
             continue
 
-        agent.base_pos = propose_position
-
         # get the horizontal distance (XZ planar projection) to the target position
-        hor_disp = agent.base_pos - target_position
+        hor_disp = propose_position - target_position
```

Comment:
Nit: I would rename `propose_position` to something along the lines of `candidate_navmesh_position`.